### ISSUE TITLE
replace fastai with pytorch-lightning

### ISSUE THREAD

**Issue by leifdenby** [convml/convml_tt#16]
Because fastai is messily written and moving quickly it's quite hard to maintain `convml_tt` to keep working with more recent versions of `fastai`. I am thinking of replacing `fastai` with more traditional `pytorch` routines, but that will take significant work. This issue is to start collecting my ideas

To make this feasible I will need to find replacements for the following functionality included in fastai:

- the learning rate finder: https://github.com/davidtvs/pytorch-lr-finder
- easy transfer learning and normalisation/de-normalisation of input/output
    - this is pretty convoluted in its implementation in fastai (https://github.com/fastai/fastai/blob/master/fastai/vision/learner.py#L139), but the interface to just call `fastai.vision.create_cnn_learner(data=data, base_arch=fastai.vision.models.resnet18, loss_func=loss_func)` is quite nice. Maybe I could keep this
- easy data augmentation

**Comment by brianmapes** [comment]
That's a shame. I am starting a reading group on AI/ML and had hoped this almost-ready demo you had set up for us last February would stay almost-ready! Fastai is fastandlooseai I guess.

**Comment by leifdenby** [comment]
Yes, I still think fastai is really fantastic as a first tool for doing ML, they have thought of really good ways of structuring their applications, but once you want a slightly different input/output it's quite messy. I've you been reading about pytorch lightning (https://pytorchlightning.ai) it looks really good, the speed and flexibility of base pytorch, but best with best practices making it possible to focus on the application and many of the features from fastai I like. 

I'm teaching a workshop next week actually, which is why I'm revisiting the code :smile: This should make it much easier for people to go in and tinker with the model architecture (which is a bit hard right now). I'm putting the notes on https://github.com/leifdenby/SENSE_convml_tt (still work-in-progress), but the exercises I did will stay the same :) Would be cool if you were to use it!

**Comment by leifdenby** [comment]
These sold it to me in any case:
- pytorch lightning intro: https://www.youtube.com/watch?v=gUF6WUq0Cf4
- comparison on fastai and pytorch lightning (and another framework called pytorch ignite): https://towardsdatascience.com/pytorch-lightning-vs-pytorch-ignite-vs-fast-ai-61dc7480ad8a

**Comment by leifdenby** [comment]
This is all done now, AH!
